mobile/native/src/health.rs

```diff
@@ -22,44 +28,96 @@ pub enum ServiceStatus {
-                            }
-                            Err(_) => {
-                                tracing::error!("Sender dropped");
-                                event::publish(&EventInternal::ServiceHealthUpdate((Service::Orderbook, ServiceStatus::Unknown)));
-                                break;
-                            },
-                        }
-                    },
-                }
-            }
-        });
+    pub fn new(config: Config, runtime: &Runtime) -> (Self, Tx) {
+        let (orderbook_tx, orderbook_rx) = watch::channel(ServiceStatus::Unknown);
+
+        let config: ConfigInternal = config.into();
+
+        let mut tasks = Vec::new();
+
+        let orderbook_monitoring =
+            runtime.spawn(publish_status_updates(Service::Orderbook, orderbook_rx));
+        tasks.push(orderbook_monitoring);
+
+        let (coordinator_tx, coordinator_rx) = watch::channel(ServiceStatus::Unknown);
+
+        let coordinator_health_endpoint =
+            "http://".to_owned() + &config.http_endpoint.to_string() + "/health";
+        let check_coordinator = tokio::spawn(check_health_endpoint(
+            coordinator_health_endpoint,
+            coordinator_tx,
+        ));
+        tasks.push(check_coordinator);
+        let coordinator_monitoring =
+            runtime.spawn(publish_status_updates(Service::Coordinator, coordinator_rx));
+        tasks.push(coordinator_monitoring);
 
         (
-            Self { _monitoring_task },
+            Self { _tasks: tasks },
             Tx {
                 orderbook: orderbook_tx,
             },
         )
     }
 }
+
+/// Publishes the health status updates for a given service to the event hub
+async fn publish_status_updates(service: Service, mut rx: watch::Receiver<ServiceStatus>) {
+    loop {
+        select! {
+            result = rx.changed() => {
+                match result {
+                    Ok(()) => {
+                        let status = rx.borrow();
+                        event::publish(&EventInternal::ServiceHealthUpdate((service, *status)));
+                    }
+                    Err(_) => {
+                        tracing::error!("Sender dropped");
+                        event::publish(&EventInternal::ServiceHealthUpdate((service, ServiceStatus::Unknown)));
+                        break;
+                    },
+                }
+            },
+        }
+    }
+}
+
+/// Periodically checks the health of a given service and updates the watch channel
+async fn check_health_endpoint(endpoint: String, tx: watch::Sender<ServiceStatus>) {
+    loop {
+        let status = if send_request(&endpoint).await.is_ok() {
+            ServiceStatus::Online
+        } else {
+            ServiceStatus::Offline
+        };
+
+        // Update the health status on the watch channel
+        tx.send(status).expect("Receiver not to be dropped");
```

Comment:
Where are you checking the `status` right now? I can't seem to find it. 

What do you think of checking the `status` in here and only forward if the service was available or not?